File: api/CS_WebShop_Api/CS_WebShop_Api/Domains/Product.cs
using System.ComponentModel.DataAnnotations;

namespace CS_WebShop_Api.Domains
{
    public class Product
    {
        [Key]
        [Required]
        public string RefNum { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Description { get; set; }
        [Required]
        public string Category { get; set; }
        [Required]
        public double Price { get; set; }
    }
}

File: web/Angular_WebShop_Web/src/modules/models/product.ts
export class Product {
    private _title: String;
    public get title(): String {
        return this._title;
    }

    private _description: String;
    public get description(): String {
        return this._description;
    }

    private _price: number;
    public get price(): number {
        return this._price;
    }

    constructor(title: String, price: number, description: String) {
        this._title = title;
        this._description = description;
        this._price = price;
    }
}